src/iam/certhandler/certhandler.cpp

```diff
@@ -155,33 +153,69 @@ Error CertHandler::GetCertificate(
     return ErrorEnum::eNone;
 }
 
-Error CertHandler::CreateSelfSignedCert(const String& certType, const String& password)
+Error CertHandler::SubscribeCertChanged(const String& certType, CertReceiverItf& certReceiver)
 {
-    LockGuard lock(mMutex);
+    LockGuard lock {mMutex};
 
-    LOG_DBG() << "Create self signed cert: type = " << certType;
+    LOG_DBG() << "Subscribe certificate receiver: type=" << certType;
 
     auto* module = FindModule(certType);
     if (module == nullptr) {
         return AOS_ERROR_WRAP(ErrorEnum::eNotFound);
     }
 
-    auto err = module->CreateSelfSignedCert(password);
+    CertInfo certInfo;
+
+    auto err = module->GetCertificate(Array<uint8_t>(), Array<uint8_t>(), certInfo);
+    if (!err.IsNone()) {
```

Comment:
Consider this change
```suggestion
    if (!err.IsNone() && !err.Is(ErrorEnum::eNotFound)) {
```

Reply:
agreed not to do that